packages/navie/src/services/project-info-service.ts

```diff
@@ -20,6 +20,25 @@ export default class ProjectInfoService {
 
     const projectInfo = Array.isArray(response) ? response : [response];
     this.interactionHistory.log('Project info obtained');
+    return projectInfo;
+  }
+
+  promptProjectInfo(isArchitecture: boolean, projectInfo: ProjectInfo[]) {
+    const emitProjectInfo: ProjectInfo[] = projectInfo.map((info) => ({ ...info }));
+
+    const isLargeProject = (appmapStats: AppMapStats) =>
+      appmapStats.packages?.length > 20 ||
+      appmapStats.routes?.length > 20 ||
+      appmapStats.tables?.length > 20;
+
+    if (!isArchitecture) {
+      for (const info of emitProjectInfo) {
+        if (info.appmapStats && isLargeProject(info.appmapStats))
+          info.appmapStats = {
+            numAppMaps: info.appmapStats.numAppMaps,
+          } as AppMapStats;
```

Comment:
This violates the type definition for `AppMapStats`. Downstream consumers may rightfully expect `packages` / `routes` / `tables` to be valid arrays.